StRoot/StFcsWaveformFitMaker/StFcsPulseAna.cxx

```diff
@@ -0,0 +1,813 @@
+  StFcsPulseAna::FillAdc(gdata,Size,Start,AdcData);
+  //for( UShort_t iprint=0; iprint<8; ++iprint ){ std::cout << " - A|StFcsPulseAna::Dep0::Data["<<iprint<<"]"<<":"<<AdcData[iprint] << std::endl; }
+  if( Size==0 ){return 0;}
+  
+  //double ped = Baseline();
+  int sum = 0 ;
+  int peak = 0 ;
+  int last = 0 ;
+  
+  for(int tb=0;tb<8;tb++) {
+
+    unsigned short radc = AdcData[tb] ;
+    
+    switch(tb) {
+    case 0 :
+      last = radc ;
+      sum = radc ;
+      peak = 0 ;
+      break ;
+    case 1 :
+      if(radc>sum) peak = 1 ;
+      sum += radc ;
+      last = radc ;
+      break ;
+    case 2 :
+      sum += radc ;
+      last = radc ;
+      break ;
+    case 3 :
+      sum += radc ;
+      last = radc ;
+      break ;
+    case 4 :
+      sum += radc ;
+      last = radc ;
+      break ;
+    case 5 :
+      sum += radc ;
+      last = radc ;
+      break ;
+    case 6 :
+      sum += radc ;
+      last = radc ;
+      break ;
+    case 7 :
+    default :
+      //printf("radc %d, last %d, peak %d\n",radc,last,peak) ;
+      
+      if(radc>=last && peak==0) {
+	sum = 0 ;
+      }
+      else {
+	sum += radc ;
+	
+	sum -= ped*8 ;	// ped is now only 3*ch_ped!
+	if(sum < 0) sum = 0 ;
+	/*
+	if(sum>0 && fcs_trgDebug>=2){ 
+	  if(geo.det<2){
+	    printf("ns=%1d det=%1d dep=%2d ch=%2d sum=%6d gain=%6d s*g=%6d pT=%8.5f\n",
+		   geo.ns,geo.det,geo.dep,geo.ch,
+		   sum,pg->gain,(sum*pg->gain)>>8,
+		   0.00024711*((sum*pg->gain)>>8) );
+	  }else{
+	    printf("ns=%1d det=%1d dep=%2d ch=%2d sum=%6d gain=%6d s*g=%6d MIP=%5.3f\n",
+		   geo.ns,geo.det,geo.dep,geo.ch,
+		   sum,pg->gain,(sum*pg->gain)>>8,
+		   float((sum*pg->gain)>>8)/100.0);
+	  }					
+	}
+	
+	sum *= pg?pg->gain:0x100 ; // note that in FY21+ gain==1.0 is 0x100
+	sum >>= 8 ;		   // see note above
+	*/
+      }
+      
+      break ;
+    }
+  }
+  return sum;
+  
+}
+
+int StFcsPulseAna::SumDep0Mod(TGraphAsymmErrors* gdata, int Start, int ped){
+  unsigned short Size = 0;//Must start with zero size
+  unsigned short AdcData[8] = {0};//Must be 8 timebin array  
+  StFcsPulseAna::FillAdc(gdata,Size,Start,AdcData);
+  if( Size==0 ){return 0;}
+
+  //double ped = Baseline();
+  int sum = 0 ;
+  int peak = 0 ;
+  int last = 0 ;
+  //int rightshift = (8*sizeof(int)-1);
+  
+  for(int tb=0;tb<8;tb++) {
+
+    unsigned short radc = AdcData[tb] ;
+    
+    switch(tb) {
+    case 0 :
+      last = radc ;
+      sum = radc ;
+      peak = 0 ;
+      break ;
+    case 1 :
+      //std::cout << "    - SumDep0Mod::Case1|peak:"<<peak << "|radc:"<<radc<<"|sum:"<<sum << std::endl;
+      if(radc>sum) peak = 1 ;
+      sum += radc ;
+      last = radc ;
+      break ;
+    case 2 :
+      //if( peak>0 && radc<=last) peak ^= 1<<rightshift ;
+      //std::cout << "    - SumDep0Mod::Case2|peak:"<<peak << "|radc:"<<radc<<"|last:"<<last << std::endl;
+      //if( peak!=0 || radc<last) peak = -1 ;
```

Comment:
I removed what I could. Please let me know if there is more that I missed. Long comments describing what the algorithms do and how they work I left in since I don't know where else to put them.